daemon/cluster/cluster.go

```diff
@@ -159,7 +167,7 @@ func New(config Config) (*Cluster, error) {
 		return nil, err
 	}
 
-	n, err := c.startNewNode(false, st.LocalAddr, st.RemoteAddr, st.ListenAddr, st.AdvertiseAddr, "", "")
```

Comment:
Happy to examine it.

- LocalAddr: The node's local address. Needed for setting up a tunnel.
- RemoteAddr: The address passed of an existing node passed to `swarm join`.
- ListenAddr: The bind address. This is usually 0.0.0.0 or `::`, so it is not redundant with `LocalAddr`.
- AdvertiseAddr: Externally-reachable address that's advertised to other nodes. Needed in one-to-one NAT scenarios.